RELEASE/scripts/autoscend/quests/level_09.ash

```diff
@@ -785,33 +786,33 @@ boolean L9_oilPeak()
 
 	if (contains_text(visit_url("place.php?whichplace=highlands"), "fire3.gif"))
 	{
-		if (!auto_is_valid($item[Bubblin\' Crude]))
+		int oilProgress = get_property("twinPeakProgress").to_int();
+		boolean needJar = ((oilProgress & 4) == 0) && item_amount($item[Jar Of Oil]) == 0;
+		if (!needJar)
 		{
 			return false;
 		}
-		int oilProgress = get_property("twinPeakProgress").to_int();
-		boolean needJar = ((oilProgress & 4) == 0);
-
-		if((item_amount($item[Bubblin\' Crude]) >= 12) && needJar)
+		else if (item_amount($item[Bubblin' Crude]) >= 12)
 		{
-			if(auto_my_path() == "G-Lover")
+			if (auto_my_path() == "G-Lover")
 			{
-				if(item_amount($item[Crude Oil Congealer]) == 0)
+				if (item_amount($item[Crude Oil Congealer]) < 1 && item_amount($item[G]) > 3)
```

Comment:
Are you intentionally saving an extra G for something? otherwise it should be `>= 3` rather than just `> 3`
as it costs 3 exactly to buy

Reply:
None of this matters. `glover_initializeSettings()` buys the crude oil congealer and that's run before even the doTasks() loop.